fix: use useRef for component key counter to prevent duplicate keys

JimStenstrom opened the pull request:
## Summary
- Fixes stale closure issue causing duplicate React keys during rapid tool execution
- Changes `componentKeyCounter` from `useState` to `useRef` 
- Exposes `getNextComponentKey()` function that synchronously increments counter

## Root Cause
The `addToChatQueue` callback captured the `componentKeyCounter` state value at creation time. When called rapidly (e.g., multiple tools executing), each call used the same stale counter value, generating duplicate keys.

## Changes
- `useAppState.tsx`: Core fix using `useRef` + `getNextComponentKey()` function
- `app.ts`, `types.ts`: Updated interface from `number` to `() => number`
- All consumers updated to call `getNextComponentKey()` instead of using number
- All test files updated with mock functions

## Test plan
- [x] Run `pnpm test:all` - all tests pass (except pre-existing read-file test failure)
- [ ] Start app and execute multiple tool calls rapidly
- [ ] Verify no "duplicate key" warnings in console

Fixes #216

Avtrkrb commented:
> Hey @JimStenstrom - thanks for this PR :) Can you fix the merge conflicts and, on sending sending a message, the CLI is exiting for some reason. See the screenshot. This also appears to clear all my prompt history.
> <img alt="Screenshot 2025-12-26 at 12 41 24" width="883" height="617" src="https://private-user-images.githubusercontent.com/89926355/530360188-0184b113-618e-4fbd-aed2-e0b49206b688.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY3NjIyNDUsIm5iZiI6MTc2Njc2MTk0NSwicGF0aCI6Ii84OTkyNjM1NS81MzAzNjAxODgtMDE4NGIxMTMtNjE4ZS00ZmJkLWFlZDItZTBiNDkyMDZiNjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjI2VDE1MTIyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2N2FiNWIxY2IwZjNjZTUzOTQ2NDViYjU1NjExMWU1ODJmZjI3MTU1NWUyZjBmZDgxMGNkMWQ2YWZjMTliZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zvLwqR8sEIL3JdHOH-ozlnA-nSneiu2Ae7p93t7TkZY">

I encountered the exact same error while testing PR #215 . I entered the `/model` command & the application abruptly exited. Nothing in the log files though.

JimStenstrom commented:
## Fix for the crash/exit issue

Found and fixed the root cause of the abrupt exit when using `/model`, `/setup-config`, or other modal commands.

### Root Cause
The refactor in PR #173 accidentally made `ChatInterface` conditionally unmount when modal modes activate. This caused ink's `Static` component to be destroyed and recreated, triggering massive memory growth (4GB in ~14 seconds) and the abrupt exit.

The code was:
```tsx
{isAnyModalMode && <ModalSelectors ... />}
{!isAnyModalMode && <ChatInterface ... />}  // Unmounts when modal active!
```

### Fix
`ChatInterface` is now always rendered. Added `hideInput` prop to hide the input area during modal mode instead of unmounting the entire component:
```tsx
<ChatInterface hideInput={isAnyModalMode} ... />  // Always mounted
{isAnyModalMode && <ModalSelectors ... />}
```

### Also Added
Debug logging throughout the render pipeline (gated behind `NANOCODER_LOG_LEVEL=debug`) to help diagnose future issues.

### Future Improvement
Created issue #227 to properly separate `ChatHistory` and `ChatInput` components, which would prevent this class of bug entirely.

will-lamerton commented:
hey @JimStenstrom - running better this end :) could you resolve conflicts and then we should be good to go!

JimStenstrom commented:
@will-lamerton, ready to merge
